docs/src/starknet/calldata-transformation.md

```diff
@@ -0,0 +1,202 @@
+
+It's declared on Sepolia network with class hash `0x02a9b456118a86070a8c116c41b02e490f3dcc9db3cad945b4e9a7fd7cec9168`.
+
+It has a few methods accepting different types and items defined in its namespace:
+
+```rust
+// data_transformer_contract/src/lib.cairo
+
+pub struct SimpleStruct {
+    a: felt252
+}
+
+pub struct NestedStructWithField {
+    a: SimpleStruct,
+    b: felt252
+}
+
+pub enum Enum {
+    One: (),
+    Two: u128,
+    Three: NestedStructWithField
+}
+
+#[starknet::contract]
+pub mod DataTransformerContract {
+    /* ... */
+
+    use super::*;
+
+    fn tuple_fn(self: @ContractState, a: (felt252, u8, Enum)) { ... }
+
+    fn nested_struct_fn(self: @ContractState, a: NestedStructWithField) { ... }
+
+    fn complex_fn(
+        self: @ContractState,
+        arr: Array<Array<felt252>>,
+        one: u8,
+        two: i16,
+        three: ByteArray,
+        four: (felt252, u32),
+        five: bool,
+        six: u256
+    ) {
+        ...
+    }
+}
+```
+
+A default form of calldata passed to commands requiring it is a series of hex-encoded felts:
+
+```shell
+$ sncast --account myuser \
+    call \
+    --url http://127.0.0.1:5050 \
+    --contract-address 0x016ad425af4585102e139d4fb2c76ce786d1aaa1cfcd88a51f3ed66601b23cdd \
+    --function tuple_fn \
+    --calldata 0x10 0x3 0x0 \
+    --block-id latest
+```
+
+However, `sncast` allows passing the data in far more handy, human-readable form - as a series of Cairo expressions.
+When calldata is delivered in such form, Cast will perform serialization automatically, based on an ABI of the contract we interact with.
+
+> 📝 **Note**
+> Calldata transformation becomes engaged when following requirements are met:
+>
+> * **Arguments are comma-separated** (on a contrary to a standard serialized input which is whitespace-separatd). Note the lack of commas in the example above - it signals Cast that the input is serialized.
+>
+> * If **only one argument without a comma** is given, it should be a **nontrivial** expression (e.g. a struct or enum constructor).\
+>  It cannot be a simple numeric litral - `--calldata 0x2137` will be treated as serialized.\
+>  To enable transformation of such a literal, we need to add a type suffix -\
+>  for example: `--calldata 0x2137_u96`. See [Caveats](./calldata-transformation.md#caveats)
+>
+> * Expressions should match the **Cairo syntax** and use **allowed items** - see [Supported expressions](./calldata-transformation.md#supported-expressions)
+
+### Basic example
+
+We can write the same command as above, but with expression calldata:
+
+```shell
+$ sncast --account myuser \
+    call \
+    --url http://127.0.0.1:5050 \
+    --contract-address 0x016ad425af4585102e139d4fb2c76ce786d1aaa1cfcd88a51f3ed66601b23cdd \
+    --function tuple_fn \
+    --calldata (0x10, 3, data_stransformer_contract::Enum::One) \
+    --block-id latest
+```
+
+getting the same result.
```

Comment:
```suggestion
getting the same result. Note that values are separated by commas `,`.
```

I know this is repeated but I think it might be worth mentioning again.